api/src/test/java/org/openmrs/notification/AlertServiceTest.java

```diff
@@ -9,14 +9,18 @@
  */
 package org.openmrs.notification;
 
+import org.junit.After;
 import org.junit.Assert;
 import org.junit.Test;
 import org.openmrs.api.context.Context;
 import org.openmrs.notification.impl.AlertServiceImpl;
 import org.openmrs.test.BaseContextSensitiveTest;
 
 public class AlertServiceTest extends BaseContextSensitiveTest {
-	
+	@After
+	public void makeSureCleanDatabase() {
+		 deleteAllData();
```

Comment:
What happens if you do not have this makeSureCleanDatabase method?

Reply:
Hey I'm not sure why this behaves like this. But it was discussed in this thread. https://talk.openmrs.org/t/switch-to-daemon-user-trunk-5189/16030/5 

If you remove the above method, the proceeding tests fail. Its like the Threads make dirty commits. Any ideas?